agent_packager/packager.py

```diff
@@ -273,7 +273,18 @@ def get_os_props():
     if HAS_DISTRO:
         return distro.name(), distro.codename()
     data = platform.dist()
-    return data[0], data[2]
+    # Sometimes the code name for distro is returned as '' and we need to
+    # fallback to the version number and pick the major version number
+    # Usually this is happened when run on some Centos 8 versions
+    # >> > import platform
+    # >> > platform.dist()
+    # ('centos', '8.3.2011', '')
+    name = data[0]
```

Comment:
this is probably not enough, because if distro is installed, then the `if HAS_DISTRO:` part will return just `.codename()`, which is... what?

Reply:
codename referes to the `Core` or `Final` in case of centos and `Santiago`/`Maipo` in case of Redhat so will  handle the HAS_DISTRO as well